packages/@aws-cdk/aws-codecommit/lib/pipeline-action.ts

```diff
@@ -31,6 +28,16 @@ export interface PipelineSourceProps extends codepipeline.CommonActionProps {
     pollForSourceChanges?: boolean;
 }
 
+/**
+ * Construction properties of the {@link PipelineSource CodeCommit source CodePipeline Action}.
+ */
+export interface PipelineSourceProps extends CommonPipelineSourceProps, codepipeline.CommonActionProps {
```

Comment:
Avoid multiple inheritance if possible (I think `CommonPiplineSourceProps` should just extend `CommonActionProps`)

Reply:
`CommonPiplineSourceProps` _cannot_ extend `CommonActionProps`, because `CommonActionProps` contain the required `stage` attribute, which we don't need to provide when calling `Repository#addToPipeline(Stage, name: string, props: CommonPiplineSourceProps)`.

I don't believe it's possible to avoid multiple inheritance here.